crates/astria-sequencer-relayer/src/relayer/celestia_client/mod.rs

```diff
@@ -124,11 +141,11 @@ impl CelestiaClient {
     /// used to obtain the appropriate fee in the case that the previous attempt failed due to a
     /// low fee.
     // Copied from https://github.com/celestiaorg/celestia-app/blob/v1.4.0/x/blob/payforblob.go
-    pub(super) async fn try_submit(
-        mut self,
+    pub(super) async fn try_prepare(
+        &mut self,
         blobs: Arc<Vec<Blob>>,
-        last_error_receiver: watch::Receiver<Option<TrySubmitError>>,
-    ) -> Result<u64, TrySubmitError> {
+        maybe_last_error: Option<TrySubmitError>,
+    ) -> Result<(BlobTxHash, BlobTx), TrySubmitError> {
```

Comment:
Mentioned this elsewhere - is it necessay for this method to return the hash? Could it be calculated later?

Reply:
Done in [9ade24c](https://github.com/astriaorg/astria/pull/1234/commits/9ade24cf8a1b470ef1ba8524afad2fcbab2dd4ab).